### ISSUE TITLE
Channel connections should close when server connection goes down

### ISSUE THREAD

**Issue by mpaolini** [Polyconseil/aioamqp#20]
Currently when protocol `connection_lost` is called by `asyncio`, no channel listeners notice it. If a greenlet is waiting on a `channel.consume()` it will be stuck there for good.

I am noticing this in our staging environment. We sometimes see:
```
[WARNING aioamqp.protocol protocol: MainProcess] Connection lost exc=ConnectionResetError(104, 'Connection reset by peer')
[INFO aioamqp.protocol protocol: MainProcess] Close connection
```

And after this log message, the program keeps waiting forever on a `channel.consume()` but it does not receive messages anymore.

I guess it could be nice to propagate the exceptions to all open channels, a bit like the `server_channel_close` in `Channel` does.

If that's ok for you, I could try and submit a patch.

**Comment by dzen** [comment]
Hi,

I think it's time to resurect this issue. Do you still have errors ?

**Comment by mpaolini** [comment]
Hmmm I think I solved this by having a janitor greenlet waiting on `protocol.wait_closed()` . When this happens, the janitor calls `cancel()` on all the other coroutines (I keep all them in a `set`).

**Comment by dzen** [comment]
did you tried using the error_callback parameter  in the connect() function ?

**Comment by RemiCardona** [comment]
I think this is fixed since PR #33 (which closes channels when the main connection goes down).
